app/models/metrics.rb

```diff
@@ -7,95 +7,57 @@ module Group
+      index: 8)
 
-    CallsReceived = MetricSorter.new(Items::CallsReceived, name: 'calls received', keypath: [:calls_received, :total])
-    CallsReceivedPerformTransaction = MetricSorter.new(Items::CallsReceivedPerformTransaction, name: 'calls received (perform transaction)', keypath: [:calls_received, :perform_transaction])
-    CallsReceivedGetInformation = MetricSorter.new(Items::CallsReceivedGetInformation, name: 'calls received (get information)', keypath: [:calls_received, :get_information])
-    CallsReceivedChaseProgress = MetricSorter.new(Items::CallsReceivedChaseProgress, name: 'calls received (chase progress)', keypath: [:calls_received, :chase_progress])
-    CallsReceivedChallengeADecision = MetricSorter.new(Items::CallsReceivedChallengeADecision, name: 'calls received (challenge a decision)', keypath: [:calls_received, :challenge_a_decision])
-    CallsReceivedOther = MetricSorter.new(Items::CallsReceivedOther, name: 'calls received (other)', keypath: [:calls_received, :other])
+    CallsReceived = MetricSortAttribute.new('calls-received', name: 'calls received', keypath: [:calls_received, :total], index: 9)
+    CallsReceivedPerformTransaction = MetricSortAttribute.new('calls-received-perform-transaction', name: 'calls received (perform transaction)', keypath: [:calls_received, :perform_transaction], index: 10)
+    CallsReceivedGetInformation = MetricSortAttribute.new('calls-received-get-information', name: 'calls received (get information)', keypath: [:calls_received, :get_information], index: 11)
+    CallsReceivedChaseProgress = MetricSortAttribute.new('calls-received-chase-progress', name: 'calls received (chase progress)', keypath: [:calls_received, :chase_progress], index: 12)
+    CallsReceivedChallengeADecision = MetricSortAttribute.new('calls-received-challenge-a-decision', name: 'calls received (challenge a decision)', keypath: [:calls_received, :challenge_a_decision], index: 13)
+    CallsReceivedOther = MetricSortAttribute.new('calls-received-other', name: 'calls received (other)', keypath: [:calls_received, :other], index: 14)
 
-    ALL = [
-      Name,
-      TransactionsReceived,
-      TransactionsReceivedOnline,
-      TransactionsReceivedPhone,
-      TransactionsReceivedPaper,
-      TransactionsReceivedFaceToFace,
-      TransactionsReceivedOther,
-      TransactionsEndingInOutcome,
-      TransactionsEndingInOutcomeWithIntendedOutcome,
-      CallsReceived,
-      CallsReceivedPerformTransaction,
-      CallsReceivedGetInformation,
-      CallsReceivedChaseProgress,
-      CallsReceivedChallengeADecision,
-      CallsReceivedOther,
-    ].freeze
+    def get_metric_sort_attribute(identifier)
+      get_all_metric_sort_attributes.select { |obj| obj.identifier == identifier }.first || Name
+    end
 
-    def self.fetch(identifier)
-      ALL.index_by(&:identifier).fetch(identifier)
+    def get_all_metric_sort_attributes
+      ObjectSpace.each_object(MetricSortAttribute).sort_by(&:index)
```

Comment:
This seems a little magical to me, and prone to breakage (for example if we were to compose a `MetricSortAttribute` from another one).

As far as I understand it, the benefit to this is it prevents us from having to explicitly list the `MetricSortAttribute`s. If we do want to accomplish that I think I'd prefer to have a "class instance-variable" on `MetricSortAttribute` which can be appended to on `initialize`, this'd make it more explicit.

However, I remain unconvinced that the implicit version outweighs the explicit in this scenario.